scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,222 @@
+        shutil.rmtree(LOGFOLDER)
+
+    # Make a new log folder
+    os.mkdir(LOGFOLDER)
+
+    # Open the output file
+    with open(LOGFOLDER + OUTPUT_FILE, "w+") as out_file:
+        # Run oc adm must-gather with the appropriate image and dest-dir
+        print("Running must-gather")
+        subprocess.run(
+            ["oc", "adm", "must-gather",
+            "--image=" + image, "--dest-dir=" + LOGFOLDER],
+            stdout=out_file)
+    
+    # Recursively walk the log folder
+    print("Searching for files to trim")
+    for subdir, _, files in os.walk(LOGFOLDER):
+        for file in files:
+            file_name = os.path.join(subdir, file)
+            with open(file_name, "r+") as curr_file:
+                # Check the number of lines in each file
+                num_lines = get_lines(curr_file)
+                if num_lines > MAX_LOGLINES:
+                    # If the maximum number of lines is too high, trim it
+                    print("Trimming %s because it exceeds %d lines"
+                        % (os.path.join(subdir, file), MAX_LOGLINES))
+                    trim_file(curr_file, MAX_LOGLINES)
+
+    # Create a time-stamped archive name
+    archive_name = ARCHIVE_NAME + "%s.tar.gz" % datetime.datetime.now().strftime("Y-%m-%d_%H-%M-%S")
+
+    # Add all the files in the log folder to a new archive file
+    with tarfile.open(archive_name, "w:gz") as tar:
+        print("Creating archive: " + archive_name)
+        tar.add(LOGFOLDER, arcname=os.path.basename(LOGFOLDER))
+
+    print("Preparing to send the data to " + BUGZILLA_URL)
+
+    file_data = ""
+    with open(archive_name, "rb") as data_file:
+        file_data = base64.b64encode(data_file.read())
+
+
+    # Send the data to the target URL (depending on whether using API key or not)
+    if use_api_key:
+        resp = send_data_with_api_key(args.api_key, bug_id, file_data)
+    else:
+        bugzilla_username = input("Enter Bugzilla username: ")
+        bugzilla_password = getpass(prompt="Enter Bugzilla password: ")
+        resp = send_data(bugzilla_username, bugzilla_password, bug_id, file_data)
+    resp_json = resp.json()
+
+    # Handle the potential errors
+    while "error" in resp_json:
+        # Using an api key will disable retries, so just output the error message
+        if use_api_key:
+            print(resp_json["message"])
+            break
+        # 300: invalid username or password
+        if resp_json["code"] == 300:
+            print("Incorrect username or password.")
+            bugzilla_username = input("Username (leave blank to exit): ")
+            if bugzilla_username == "":
+                print("Username left blank, exiting")
+                exit()
+            bugzilla_password = getpass(prompt="Password: ")
+            resp = send_data(bugzilla_username, bugzilla_password, bug_id, file_data)
+            resp_json = resp.json()
+        # 101: Invalid bug id
+        elif resp_json["code"] == 101:
```

Comment:
what about any other error code?

Reply:
These are the ones that as of now I can test, once I have access to somewhere I can test that attachment is working I can try to deal with other error codes.

Reply:
Please handle any other code in else block as generic issue.

Reply:
Done in recent commit; any other code will display the error that Bugzilla returns and will exit

Reply:
:+1: